core/src/lib.rs

```diff
@@ -145,89 +187,101 @@ mod tests {
-        kv_body
-            .entry("expirationDate".to_string())
-            .or_insert(Value::String("2029-12-03T12:19:52Z".to_string()));
-
-        kv_subject
-            .entry("id".to_string())
-            .or_insert(Value::String("did:example:b34ca6cd37bbf23".to_string()));
-
-        let type_rs = serde_json::to_value(["PermanentResident".to_string(), "Person".to_string()]);
-        if type_rs.is_ok() {
-            kv_subject
-                .entry("type".to_string())
-                .or_insert(type_rs.unwrap());
-        }
-
-        kv_subject
-            .entry("givenName".to_string())
-            .or_insert(Value::String("JOHN".to_string()));
-        kv_subject
-            .entry("familyName".to_string())
-            .or_insert(Value::String("SMITH".to_string()));
-        kv_subject
-            .entry("gender".to_string())
-            .or_insert(Value::String("Male".to_string()));
-        kv_subject
-            .entry("image".to_string())
-            .or_insert(Value::String(
-                "data:image/png;base64,iVBORw0KGgo...kJggg==".to_string(),
-            ));
-        kv_subject
-            .entry("residentSince".to_string())
-            .or_insert(Value::String("2015-01-01".to_string()));
-        kv_subject
-            .entry("lprCategory".to_string())
-            .or_insert(Value::String("C09".to_string()));
-        kv_subject
-            .entry("lprNumber".to_string())
-            .or_insert(Value::String("999-999-999".to_string()));
-        kv_subject
-            .entry("commuterClassification".to_string())
-            .or_insert(Value::String("C1".to_string()));
-        kv_subject
-            .entry("birthCountry".to_string())
-            .or_insert(Value::String("Bahamas".to_string()));
-        kv_subject
-            .entry("birthDate".to_string())
-            .or_insert(Value::String("1958-07-17".to_string()));
+        let (kv_body, kv_subject) = get_body_subject();
 
         let vc = to.create_credential(
             crate::CRED_TYPE_PERMANENT_RESIDENT_CARD.to_string(),
             kv_subject,
             kv_body,
             "https://issuer.oidp.uscis.gov/credentials/83627465",
         );
+
         assert!(vc.is_ok());
-        assert_json_eq!(expect, vc.unwrap());
+        let credential = vc.unwrap();
+        assert_json_eq!(expect_credential, credential.serialize());
+
+        let signer = signature::signer::Ed25519DidSigner::new();
+        let proof = create_data_integrity_proof(
+            &signer,
+            credential.serialize(),
+            signature::suite::VerificationRelation::AssertionMethod,
+        );
+
+        assert!(proof.is_ok());
+
+        let verifiable_credential = credential.create_verifiable_credentials(proof.unwrap());
+        let credentials = vec![verifiable_credential];
+        let pre = to.create_presentation(credentials);
+
+        assert!(pre.is_ok());
+
+        let interim_presentation = pre.unwrap();
```

Comment:
depending on your intent, there's a more verbose version of `.unwrap()` called [`.expect("message")`](https://doc.rust-lang.org/std/result/enum.Result.html#method.expect) that allows you to be comment on your unwrapped and panicking error:
```suggestion
        let interim_presentation = to.create_presentation(credentials).expect("unable to create presentation from credentials");
```